httpclient5-cache/src/main/java/org/apache/hc/client5/http/impl/cache/memcached/MemcachedHttpAsyncCacheStorage.java

```diff
@@ -161,7 +162,11 @@ private <T> Cancellable operation(final OperationFuture<T> operationFuture, fina
                 callback.completed(operationFuture.get());
             } catch (final ExecutionException ex) {
                 if (ex.getCause() instanceof Exception) {
-                    callback.failed((Exception) ex.getCause());
+                    if (ex.getCause() instanceof CancellationException) {
+                        callback.failed(new ResourceIOException("Cache operation was cancelled mid-flight", ex.getCause()));
```

Comment:
Now that I'm reading this, I should probably create an `OperationCancelledException` that extends `ResourceIOException` and use this for all of these...